src/main/java/edu/ucar/unidata/rosetta/repository/DataDao.java

```diff
@@ -0,0 +1,43 @@
+package edu.ucar.unidata.rosetta.repository;
+
+import edu.ucar.unidata.rosetta.domain.Data;
+import org.springframework.dao.DataRetrievalFailureException;
+
+/**
+ * @author oxelson@ucar.edu
+ */
+public interface DataDao {
+
+    /**
+     * Looks up and retrieves a persisted Data object using the given id.
+     *
+     * @param id    The id of the Data object.
+     * @return      The Data object corresponding to the given id.
+     * @throws DataRetrievalFailureException  If unable to lookup Data with the given id.
+     */
+    public Data lookupById(String id) throws DataRetrievalFailureException;
+
+    /**
+     * Persists the information in the given data object.
+     *
+     * @param data  The Data object to persist.
+     * @throws DataRetrievalFailureException  If unable to persist the Data object.
+     */
+    public void persistData(Data data) throws DataRetrievalFailureException;
+
+    /**
+     * Updated the information corresponding to the given data object.
+     *
+     * @param data  The data object to update.
+     * @throws DataRetrievalFailureException  If unable to update persisted Data object.
+     */
+    public void updatePersistedData(Data data) throws DataRetrievalFailureException;
+
+    /**
+     * Deletes the persisted data object information.
+     *
+     * @param id    The id of the Data object to delete.
+     * @throws DataRetrievalFailureException  If unable to delete persisted Data object.
+     */
+    public void deletePersistedData(String id) throws DataRetrievalFailureException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.springframework.dao.DataRetrievalFailureException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)